processor/metricstarttimeprocessor/internal/datapointstorage/timeseries_map.go

```diff
@@ -26,13 +26,15 @@ type TimeseriesInfo struct {
 }
 
 type NumberInfo struct {
-	StartTime     pcommon.Timestamp
-	PreviousValue float64
+	StartTime           pcommon.Timestamp
+	PreviousDoubleValue float64
```

Comment:
Did you consider just storing the pcommon.Value?  Seems like it would be more readable and simplify code